### PR TITLE
feat: configure built-in logging

### PR DESCRIPTION
**Pull request by kov117**
### Checklist

- [x] I've updated both the `azurerm_linux_function_app` and `azurerm_windows_function_app` resources.

### CODE REVIEW

**Comment by hknutsen** [comment]
It's recommended for built-in logging to be enabled by default: https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#disable-built-in-logging

**Comment by kov117** [comment]
will change the configuration to disabled 🫡

**Comment by hknutsen** [comment]
> It's recommended for built-in logging to be **disabled** by default: https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#disable-built-in-logging

Could even consider hard-coding it to be disabled, and recommend using App Insights instead. A variable for enabling built-in logging can always be added at a later date if absolutely needed.

**Comment by kov117** [comment]
i don't see a problem with that, if the rest of the team @equinor/terraform-baseline-maintainers  agree then i will implement the suggested change 👍🏼

**Comment by kov117** [comment]
Ran a new test and it passed 👍🏼

**Comment by hknutsen** [comment]
> > It's recommended for built-in logging to be **disabled** by default: https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#disable-built-in-logging
> 
> Could even consider hard-coding it to be disabled, and recommend using App Insights instead. A variable for enabling built-in logging can always be added at a later date if absolutely needed.

Waiting for feedback from @equinor/terraform-baseline-maintainers.

**Comment by hknutsen** [comment]
@kov117 Hardcode built-in logging to be **disabled** for now, and update tests accordingly. We can always add a variable later if absolutely necessary.

**Comment by kov117** [comment]
Terraform test passed 👍🏼

File: tests/logs.unit.tftest.hcl
Comment (hknutsen) [inline tests/logs.unit.tftest.hcl:1]: Do these tests not do exactly the same as the tests you've added in `defaults.unit.tftest.hcl`?

**Comment by kov117** [comment]
Ran test again just to be sure and it passed 👍🏼

**Review by hknutsen — APPROVED** [review]
Looks good 👍

**Pull request merged**
